validator/rpc/server.go

```diff
@@ -123,13 +125,24 @@ func NewServer(ctx context.Context, cfg *Config) *Server {
 		log.WithError(err).Fatal("Could not register beacon chain gRPC or HTTP client")
 	}
 
-	if err := server.InitializeRoutesWithWebHandler(); err != nil {
-		log.WithError(err).Fatal("Could not initialize routes with web handler")
+	var allowedOrigins []string
+	if cliCtx.IsSet(flags.HTTPServerCorsDomain.Name) {
+		allowedOrigins = strings.Split(cliCtx.String(flags.HTTPServerCorsDomain.Name), ",")
+	} else {
+		allowedOrigins = strings.Split(flags.HTTPServerCorsDomain.Value, ",")
 	}
 
+	middlewares := []middleware.Middleware{
+		middleware.NormalizeQueryValuesHandler,
+		middleware.CorsHandler(allowedOrigins),
+	}
 	opts := []httprest.Option{
 		httprest.WithRouter(cfg.Router),
 		httprest.WithHTTPAddr(net.JoinHostPort(server.httpHost, fmt.Sprintf("%d", server.httpPort))),
+		httprest.WithMiddlewares(middlewares),
+	}
+	if err := server.InitializeRoutesWithWebHandler(); err != nil {
```

Comment:
moved this below because this fn is calling initialize routes where I am appending a middleware to the list but I create a list in L142 thats why placed it here also no particular issue I can see with ordering here.